launch_chffrplus.sh

```diff
@@ -55,18 +55,22 @@ function launch {
     fi
   fi
 
-  # no cpu rationing for now
-  echo 0-3 > /dev/cpuset/background/cpus
-  echo 0-3 > /dev/cpuset/system-background/cpus
-  echo 0-3 > /dev/cpuset/foreground/boost/cpus
-  echo 0-3 > /dev/cpuset/foreground/cpus
-  echo 0-3 > /dev/cpuset/android/cpus
-
-  # change interrupt affinity
-  echo 3 > /proc/irq/6/smp_affinity_list # MDSS
-  echo 1 > /proc/irq/78/smp_affinity_list # Modem, can potentially lock up
-  echo 2 > /proc/irq/733/smp_affinity_list # USB
-  echo 2 > /proc/irq/736/smp_affinity_list # USB
+  # Android and other system processes are not permitted to run on CPU 3
+  # NEOS installed app processes can run anywhere
+  echo 0-2 > /dev/cpuset/background/cpus
+  echo 0-2 > /dev/cpuset/system-background/cpus
+  echo 0-2 > /dev/cpuset/foreground/boost/cpus
+  echo 0-2 > /dev/cpuset/foreground/cpus
+  echo 0-2 > /dev/cpuset/android/cpus
+  echo 0-3 > /dev/cpuset/app/cpus
+
+  # Collect RIL and other possibly long-running I/O interrupts onto CPU 1
+  echo 1 > /proc/irq/78/smp_affinity_list # qcom,smd-modem (LTE radio)
+  echo 1 > /proc/irq/33/smp_affinity_list # ufshcd (flash storage)
+  echo 1 > /proc/irq/35/smp_affinity_list # wifi (wlan_pci)
+  # USB traffic needs realtime handling on cpu 3
+  [ -d "/proc/irq/733" ] && echo 3 > /proc/irq/733/smp_affinity_list # USB for LeEco
+  [ -d "/proc/irq/736" ] && echo 3 > /proc/irq/736/smp_affinity_list # USB for OP3T
```

Comment:
Maybe we should wrap these in an if statement so it doesn't just silently fail if for some reason both of these don't exist?

Reply:
Short term, this script really only works on NEOS, which will definitely have one of those two interrupts. Long term, I'd like to have at-launch platform detection and then have individual platform setup routines, such that we can automatically startup on Ubuntu, OSX, NEOS/EON, NEOS/LEON, NEOS/C2, any future platform, etc. That's becoming more important every day. For those scripts, we can and should fail loudly if platform mandatory stuff is missing.